test/typescript/t.test.ts

```diff
@@ -172,7 +172,7 @@ function datamap(t: TFunction) {
   interface TransDataMap {
     'Greeting': { name: string; };
   }
-  function trans<K extends keyof TransDataMap>(key: K, data: TransDataMap[K]): string {
+  function trans<K extends keyof TransDataMap>(key: K, data: TransDataMap[K]): string | TFunctionDetailedResult<string> {
```

Comment:
This has a problem that it doesn't work with `el.textContent = trans(...)`. And mismatching of the return type has a function to prevent the data type form adding invalid properties. I'll improve the test.

Reply:
What I don't understand is, why is `TOpt['returnDetails'] extends true` always true?

Reply:
Because type `TOpt['returnDetails']` is not decided. So the result of the type function is both return types as all the possibilities.

Reply:
quantum logic 😜

Reply:
Fixed.